gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -163,8 +165,14 @@ public MRJobLauncher(Properties jobProps, Configuration conf)
     this.parallelRunnerThreads = Integer.parseInt(jobProps.getProperty(ParallelRunner.PARALLEL_RUNNER_THREADS_KEY,
         Integer.toString(ParallelRunner.DEFAULT_PARALLEL_RUNNER_THREADS)));
 
+    // StateStore interface uses the following key (rootDir, storeName, tableName)
+    // The state store base is the root directory and the last two elements of the path are used as the storeName and
+    // tableName. Create the state store with the root at jobOutputPath. The task state will be stored at
+    // jobOutputPath/output/taskState.tst, so output will be the storeName.
+    taskStateStore = new FsStateStore<>(this.fs, jobOutputPath.toString(), TaskState.class);
```

Comment:
I spent some time working on this, but hit some issues with the file format change for the workunits which are not currently stored in a StateStore. Since this is not essential and likely won't be used, I would prefer to leave the code alone to avoid inadvertently introducing bugs or compatibility issues.